lib/std/json.zig

```diff
@@ -1551,8 +1551,11 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                 .Number => |n| n,
                 else => return error.UnexpectedToken,
             };
-            if (!numberToken.is_integer) return error.UnexpectedToken;
-            return try std.fmt.parseInt(T, numberToken.slice(tokens.slice, tokens.i - 1), 10);
+            if (numberToken.is_integer)
+                return try std.fmt.parseInt(T, numberToken.slice(tokens.slice, tokens.i - 1), 10);
+            const float = try std.fmt.parseFloat(f128, numberToken.slice(tokens.slice, tokens.i - 1));
```

Comment:
Json numbers are `f64`, why is `f128` used here?
I'd expect the function to respect the user-specified type `T` and try parsing the number as a fp-type with the same number of bits, not `f128`.

Reply:
> Json numbers are `f64`, why is `f128` used here?

huh? json numbers are not specified to have any precision: many people use them to arbitrary precision.

Reply:
JSON uses doubles for numbers, the exponential notation means the string is formatted as a floating point number hence the limited precision.
AFAIK JSON doesnt mandate the use of arbitrarily big fp types.